REGRESSION (281723@main):[ iOS ]: 22 API test are constantly crashing.

xeenon opened the pull request:
#### 1f1720a65d5655edf8cc18ff247acd9ceee0363c
<pre>
REGRESSION (281723@main):[ iOS ]: 22 API test are constantly crashing.
<a href="https://webkit.org/b/277563">https://webkit.org/b/277563</a>
<a href="https://rdar.apple.com/problem/133081797">rdar://problem/133081797</a>

Reviewed by Wenson Hsieh.

Define the old Web Extension API symbols to maintain bincompat with older OS builds.

* Source/WebKit/SourcesCocoa.txt: Added _WKWebExtension.mm.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.h: Added @interface.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.mm: Added.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionAction.h: Added @interface.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionCommand.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionController.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionDataRecord.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionMatchPattern.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionMessagePort.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionTabCreationOptions.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionWindowCreationOptions.h: Ditto.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj: Added _WKWebExtension.mm.

Canonical link: <a href="https://commits.webkit.org/281809@main">https://commits.webkit.org/281809@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9f0671a741202c00f3b04ab0d2f9ed021d6bd9af

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/61058 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/40419 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13639 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64995 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11607 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/48096 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11882 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/49359 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/60/builds/8063 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/63092 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37615 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/52898 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30189 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/34294 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/10519 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56112 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/10418 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66725 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/5005 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/10233 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/56724 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/5027 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52861 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56917 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4146 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9184 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/36223 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/37306 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/38400 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/37050 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

AriYoung00 commented on Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.mm:
```diff
@@ -0,0 +1,121 @@
+WK_EXTERN NSNotificationName const _WKWebExtensionErrorsWereUpdatedNotification = WKWebExtensionErrorsWereUpdatedNotification;
+
+WK_EXTERN
+@interface _WKWebExtension : WKWebExtension
```
Just for my own edification -- Is there any particular reason to use an empty inheriting interface to define these symbols rather than `@compatibility_alias`?

whsieh replied:
Great catch! I think that should work as well (and would be a lot cleaner, assuming it does).

AriYoung00 replied:
After a cursory Slack search it seems like this might just affect headers, and not what symbols are actually exported during build -- so not sure if this will work for binary compatibility purposes.

whsieh replied:
I see, thanks for double checking!

xeenon replied:
I could have used that instead of `#define` though! I'll keep that in mind for next time.

whsieh reviewed (approved):
Is there a reason this is compiled directly as a source in the Xcodeproj file list, rather than unified sources?

xeenon committed:
REGRESSION (281723@main):[ iOS ]: 22 API test are constantly crashing.
https://webkit.org/b/277563
rdar://problem/133081797

Reviewed by Wenson Hsieh.

Define the old Web Extension API symbols to maintain bincompat with older OS builds.

* Source/WebKit/SourcesCocoa.txt: Added _WKWebExtension.mm.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.h: Added @interface.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.mm: Added.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionAction.h: Added @interface.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionCommand.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionController.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionDataRecord.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionMatchPattern.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionMessagePort.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionTabCreationOptions.h: Ditto.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionWindowCreationOptions.h: Ditto.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj: Added _WKWebExtension.mm.

Canonical link: https://commits.webkit.org/281809@main
